### PR TITLE
feat: Add AI Model Config resource and data source

### PR DESCRIPTION
**Pull request by tracisiebel**
## Summary

Adds support for AI Model Config resources and data sources to the LaunchDarkly Terraform provider. AI Model Configs allow users to configure LLM model settings including provider, parameters, and pricing information.

**Link to Devin run**: https://app.devin.ai/sessions/e41fa8370ea042beb14d4de2a208f98c  
**Requested by**: @tracisiebel (traci@launchdarkly.com)

### New Resources
- **Resource**: `launchdarkly_ai_model_config` - Create and manage AI model configurations
- **Data Source**: `launchdarkly_ai_model_config` - Read existing AI model configurations

### Key Features
- Support for all AI model config fields: `model_id`, `model_provider`, `icon`, `params`, `custom_params`, `cost_per_input_token`, `cost_per_output_token`, `tags`
- **Immutable resources**: AI model configs cannot be updated after creation (enforced via `CustomizeDiff`)
- Comprehensive test coverage including immutability tests for all fields

### SDK Upgrade
- Upgraded `api-client-go` from v17.1.0 to v17.2.0 (required for AI Model Config GA support)
- Fixed v17.2.0 breaking changes:
  - `flag.Environments` is now `*map[string]ldapi.FeatureFlagConfig` (pointer to map)
  - `MetricPost.IsActive` field removed (can only be set via PATCH after creation)

### Beta Client Cleanup
- Removed beta client configuration (`ldBeta`, `newBetaClient`) since AI Config APIs are now GA
- Updated all AI Model Config API calls to use standard client without beta headers
- This addresses the DRY concern raised in review by eliminating duplicate configuration functions

## Testing

✅ **Comprehensive test coverage added**:
- Basic CRUD operations
- All attributes tested (required and optional)
- Import/export functionality
- Force new on key/project_key changes
- Immutability enforcement for: `name`, `model_id`, `tags`, `params`, `custom_params`, `cost_per_input_token`, `cost_per_output_token`
- Data source functionality

⚠️ **Note**: Acceptance tests have not been run yet - CI is currently running.

## Human Review Checklist

**Critical items to review**:

1. **SDK upgrade side effects**: The v17.2.0 upgrade has breaking changes. Verify that:
   - All `flag.Environments` pointer dereferencing is correct (changed in test files)
   - Removing `IsActive` from `MetricPost` doesn't break metric creation
   - No other v17.2.0 breaking changes were missed

2. **Beta client removal impact**: I removed `newBetaClient()` and updated metric tests to use standard client. Verify that:
   - Experimentation endpoints don't require beta headers
   - Metric tests with randomization units still work correctly

3. **Immutability implementation**: Verify that:
   - All fields that should be immutable are covered in `CustomizeDiff`
   - The error messages are clear and helpful
   - The immutability tests cover all edge cases

4. **Documentation**: This PR does not include documentation updates. Should we add:
   - Resource documentation for `launchdarkly_ai_model_config`
   - Data source documentation for `launchdarkly_ai_model_config`
   - Examples showing typical usage

5. **API naming**: The API service is still called `AIConfigsBetaApi` in the SDK even though the APIs are GA. This is just a naming artifact and shouldn't cause issues, but worth noting.

## Known Limitations

- AI Model Config resources are **immutable** - any changes require destroying and recreating the resource
- This is by design based on the API behavior

## Follow-up Items

- [ ] Add documentation for the new resource and data source
- [ ] Verify all acceptance tests pass in CI
- [ ] Consider adding examples to the documentation

### CODE REVIEW

File: launchdarkly/ai_model_config_helper.go [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+			Type:        schema.TypeString,
+			Required:    !isDataSource,
+			Computed:    isDataSource,
+			ForceNew:    !isDataSource,
+			Description: addForceNewDescription("The human-friendly name for the AI model config.", !isDataSource),
+		},
+		ID: {
+			Type:        schema.TypeString,
+			Required:    !isDataSource,
+			Computed:    isDataSource,
```
Comment (sloloris) [inline launchdarkly/ai_model_config_helper.go]: does the user assign this or does LD? if LD assigns this it should always be computed and the force new is moot

File: launchdarkly/ai_model_config_helper.go [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+			Optional:    !isDataSource,
+			Computed:    isDataSource,
+			ForceNew:    !isDataSource,
+			Description: addForceNewDescription("Icon for the model.", !isDataSource),
+		},
+		PARAMS: {
+			Type:        schema.TypeString,
+			Optional:    !isDataSource,
+			Computed:    isDataSource,
+			ForceNew:    !isDataSource,
```
Comment (sloloris) [inline launchdarkly/ai_model_config_helper.go]: ditto force new

File: launchdarkly/ai_model_config_helper.go [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+			Description: addForceNewDescription("Cost per output token in USD.", !isDataSource),
+		},
+		VERSION: {
+			Type:        schema.TypeInt,
+			Computed:    true,
+			Description: "Version of the AI model config.",
+		},
+		GLOBAL: {
+			Type:        schema.TypeBool,
+			Computed:    true,
```
Comment (sloloris) [inline launchdarkly/ai_model_config_helper.go]: is this user-defined or does LD auto-assign the value? if the former it should not be computed

**Review by sloloris — CHANGES_REQUESTED** [review]
hi @tracisiebel ! 

before we approve and release this, could you get someone from your team who is familiar with the resource and the API to review this, especially the schema? i also don't think we have many (if any) other resources that don't allow update / have an associated patch API - i think it is worth having a more deliberate conversation about how we actually expect the provider to behave in this situation. 

also, we will want to ensure it is thoroughly tested before we merge.

**Comment by tracisiebel** [comment]
@sloloris tested with this command:

```GOENV_VERSION=1.23.5 TF_ACC=1 go test -v ./launchdarkly/ -run "AIModelConfig" -timeout 30m```

everything passed. Added some tests.

File: launchdarkly/ai_model_config_helper.go (hunk 3 above)
Comment (tracisiebel) [reply launchdarkly/ai_model_config_helper.go]: LD auto-assigns

File: launchdarkly/resource_launchdarkly_metric.go [hunk 4]
```diff
@@ -191,7 +191,6 @@ func resourceMetricCreate(ctx context.Context, d *schema.ResourceData, metaRaw i
 	kind := d.Get(KIND).(string)
 	description := d.Get(DESCRIPTION).(string)
 	tags := stringsFromResourceData(d, TAGS)
-	isActive := d.Get(IS_ACTIVE).(bool)
```
Comment (tracisiebel) [inline launchdarkly/resource_launchdarkly_metric.go]: reverting this change

File: launchdarkly/data_source_launchdarkly_feature_flag_environment_test.go [hunk 5]
```diff
@@ -156,8 +156,11 @@ func TestAccDataSourceFeatureFlagEnvironment_exists(t *testing.T) {
 		require.NoError(t, err)
 	}()
 
-	thisConfig := flag.Environments[envKey]
-	otherConfig := flag.Environments["production"]
+	if flag.Environments == nil {
```
Comment (ldhenry) [inline launchdarkly/data_source_launchdarkly_feature_flag_environment_test.go]: I'm curious why changes to this file were needed since the AI config resource is unrelated.

**Review by ldhenry — CHANGES_REQUESTED** [review]
In general, we try not to release resources that are still behind beta APIs. Do y'all have near-term plans to take the AI config endpoints out of beta?

**Comment by ldhenry** [comment]
> @ldhenry not ready for review. need to pick up the new changes (without beta) before it's ready. I'll let you know

Sounds good. Just keep us in the loop.
